src/Lightning.ts

```diff
@@ -807,8 +807,24 @@ export const LightningMixin = (superclass) =>
         // TODO: we should not log/keep secret in the logs
         this.logger.debug({ invoice, user: this.user }, "got invoice status")
       } catch (err) {
-        const error = `issue fetching invoice`
-        this.logger.warn({ err, invoice }, error)
+        const invoiceNotFound = "unable to locate invoice"
+        try {
```

Comment:
Why is this a `try/catch` instead of an `if/else`?

Reply:
I agree this should be a if/else but the reason I did a try/catch instead if because the error being returned from the lightning library I believe is loosely defined (and currently not typed). so I'm not sure there is always an `err` property at `err[2]` and and that this `err` contains a `details` property. would need to parse the alexbosworth/lightning library to make this is the case. I have had previous issue with it in the past. (it's also an pseudo emergency PR)